engine/src/btc/rpc.rs

```diff
@@ -261,7 +261,6 @@ pub enum Difficulty {
 pub struct VerboseTransaction {
 	pub txid: Txid,
 	pub hash: Txid,
```

Comment:
I feel there should also be a comment here, stating why the Version is excluded, as we will forget.

Reply:
Keeping the field, but changing its type to something like this could be more self-explanatory:

```rust
#[derive(Clone, PartialEq, Debug, Deserialize, Serialize)]
#[serde(untagged)]
pub enum VerboseTransactionVersion {
    Valid(Version),
    Invalid(serde_json::Value),
}
```

```rust
#[test]
fn test_verbose_transaction_version() {
    for (input, expected_output) in [
        (json!(1), VerboseTransactionVersion::Valid(Version::from_consensus(1))),
        (json!(2), VerboseTransactionVersion::Valid(Version::from_consensus(2))),
        (json!(u32::MAX), VerboseTransactionVersion::Invalid(json!(u32::MAX))),
    ] {
        assert_eq!(serde_json::from_value::<VerboseTransactionVersion>(input).unwrap(), expected_output);
    }
}```

Reply:
essentialy, any u32 value is allowed. Seems that the correct way to parse this would be to parse a u32 and then reinterpret it as i32